### PR TITLE
Don't Validate Access Type during Pingback Logging

### PR DESCRIPTION
**Pull request by elijahsoria**
We want to send the pingback on all logging events, regardless to if gaa_at == na.

### CODE REVIEW

File: src/utils/gaa.js [hunk 1]
```diff
@@ -292,9 +292,13 @@ export let GoogleUserDef;
 /**
  * Returns true if the query string contains fresh Google Article Access (GAA) params.
  * @param {string} queryString
+ * @param {boolean} shouldValidateAccessType
  * @return {boolean}
  */
-export function queryStringHasFreshGaaParams(queryString) {
+export function queryStringHasFreshGaaParams(
+  queryString,
+  shouldValidateAccessType = true
```
Comment (ChrisAntaki) [inline src/utils/gaa.js:300]: Could we update this name to `allowAllAccessTypes`? I think that'd be a bit clearer

File: src/runtime/entitlements-manager.js [hunk 2]
```diff
@@ -234,7 +234,13 @@ export class EntitlementsManager {
   possiblyPingbackOnClientEvent_(event) {
     // Verify GAA params are present, otherwise bail since the pingback
     // shouldn't happen on non-metering requests.
-    if (!queryStringHasFreshGaaParams(this.win_.location.search)) {
+    // We don't validate access type since not all publishers use it.
```
Comment (ChrisAntaki) [inline src/runtime/entitlements-manager.js:237]: Could this comment read "We always want to send pingbacks" ?

Comment (elijahsoria) [reply src/runtime/entitlements-manager.js:237]: Added some more info along these lines.

File: src/utils/gaa.js (hunk 1 above)
Comment (elijahsoria) [reply src/utils/gaa.js:300]: Done

File: src/utils/gaa.js [hunk 3]
```diff
@@ -292,9 +292,13 @@ export let GoogleUserDef;
 /**
  * Returns true if the query string contains fresh Google Article Access (GAA) params.
  * @param {string} queryString
+ * @param {boolean} allowAllAccessTypes
  * @return {boolean}
  */
-export function queryStringHasFreshGaaParams(queryString) {
+export function queryStringHasFreshGaaParams(
+  queryString,
+  allowAllAccessTypes = true
```
Comment (ChrisAntaki) [inline src/utils/gaa.js:300]: Could we make this default to false now?

File: src/utils/gaa-test.js [hunk 4]
```diff
@@ -108,6 +108,11 @@ describes.realWin('queryStringHasFreshGaaParams', {}, () => {
     const queryString = '?gaa_at=na&gaa_n=n&gaa_sig=sig&gaa_ts=99999';
     expect(queryStringHasFreshGaaParams(queryString)).to.be.false;
   });
+
+  it('succeeds if gaa_at param specifies "no access" but shouldValidateAccessType is false', () => {
```
Comment (ChrisAntaki) [inline src/utils/gaa-test.js:112]:
```suggestion
  it('succeeds if gaa_at param specifies "no access" but allowAllAccessTypes is true', () => {
```

Comment (elijahsoria) [reply src/utils/gaa-test.js]: Updated

File: src/utils/gaa.js (hunk 3 above)
Comment (elijahsoria) [reply src/utils/gaa.js]: Yep!

**Comment by ChrisAntaki** [comment]
Thanks!

**Review by ChrisAntaki — APPROVED** [review]

**Pull request merged**